lib/browser.ts

```diff
@@ -837,79 +848,85 @@ export class ProtractorBrowser extends AbstractExtendedWebDriver {
       return 'Protractor.get(' + destination + ') - ' + str;
     };
 
+    let retPromise: wdpromise.Promise<any>;
+    let then = (fn: () => wdpromise.Promise<any>) => {
+      if (retPromise) {
+        return (retPromise = retPromise.then(fn));
+      } else {
+        return (retPromise = fn());
+      }
+    };
+
     if (this.bpClient) {
-      this.driver.controlFlow().execute(() => {
+      then(() => this.driver.controlFlow().execute(() => {
         return this.bpClient.setSynchronization(false);
-      });
+      }));
     }
 
     if (this.ignoreSynchronization) {
-      this.driver.get(destination);
-      return this.driver.controlFlow().execute(() => this.plugins_.onPageLoad()).then(() => {});
+      then(() => this.driver.get(destination));
+      return then(() => this.driver.controlFlow().execute(() => this.plugins_.onPageLoad()))
```

Comment:
Move the bpClient.setSynchronization call after this test. I'm not sure, but I think Circle is failing because waitForAngular gets turned off in BP but never gets restored if get() returns from ehre.